drivers/eeprom/eeprom_at2x.c

```diff
@@ -550,138 +550,62 @@ static const struct eeprom_driver_api eeprom_at2x_api = {
-		.addr_width = DT_INST_##n##_ATMEL_AT##t##_ADDRESS_WIDTH, \
-		.readonly = DT_INST_##n##_ATMEL_AT##t##_READ_ONLY, \
-		.timeout =  DT_INST_##n##_ATMEL_AT##t##_TIMEOUT, \
+		.bus_dev_name = DT_BUS_LABEL(DT_INST_AT2X(n, t)), \
+		.bus_addr = DT_REG_ADDR(DT_INST_AT2X(n, t)), \
+		.max_freq = UTIL_AND( \
+			DT_NODE_HAS_PROP(DT_INST_AT2X(n, t), \
+					 spi_max_frequency), \
+			DT_PROP(DT_INST_AT2X(n, t), spi_max_frequency)), \
+		.spi_cs_dev_name = UTIL_AND( \
+			DT_SPI_DEV_HAS_CS_GPIOS(DT_INST_AT2X(n, t)), \
+			DT_SPI_DEV_CS_GPIOS_LABEL(DT_INST_AT2X(n, t))),	\
+		.spi_cs_pin = UTIL_AND( \
+			DT_SPI_DEV_HAS_CS_GPIOS(DT_INST_AT2X(n, t)), \
+			DT_SPI_DEV_CS_GPIOS_PIN(DT_INST_AT2X(n, t))), \
+		.wp_gpio_pin = UTIL_AND( \
+			DT_NODE_HAS_PROP(DT_INST_AT2X(n, t), wp_gpios), \
+			DT_GPIO_PIN(DT_INST_AT2X(n, t), wp_gpios)), \
+		.wp_gpio_flags = UTIL_AND( \
+			DT_NODE_HAS_PROP(DT_INST_AT2X(n, t), wp_gpios), \
+			DT_GPIO_FLAGS(DT_INST_AT2X(n, t), wp_gpios)), \
+		.wp_gpio_name = UTIL_AND( \
+			DT_NODE_HAS_PROP(DT_INST_AT2X(n, t), wp_gpios), \
+			DT_GPIO_LABEL(DT_INST_AT2X(n, t), wp_gpios)), \
+		.size = DT_PROP(DT_INST_AT2X(n, t), size), \
+		.pagesize = DT_PROP(DT_INST_AT2X(n, t), pagesize), \
+		.addr_width = DT_PROP(DT_INST_AT2X(n, t), address_width), \
+		.readonly = DT_PROP(DT_INST_AT2X(n, t), read_only), \
+		.timeout = DT_PROP(DT_INST_AT2X(n, t), timeout), \
 		.read_fn = eeprom_at##t##_read, \
 		.write_fn = eeprom_at##t##_write, \
 	}; \
 	static struct eeprom_at2x_data eeprom_at##t##_data_##n; \
 	DEVICE_AND_API_INIT(eeprom_at##t##_##n, \
-			    DT_INST_##n##_ATMEL_AT##t##_LABEL, \
+			    DT_LABEL(DT_INST_AT2X(n, t)), \
 			    &eeprom_at2x_init, &eeprom_at##t##_data_##n, \
 			    &eeprom_at##t##_config_##n, POST_KERNEL, \
 			    CONFIG_KERNEL_INIT_PRIORITY_DEVICE, \
 			    &eeprom_at2x_api)
 
-#ifdef CONFIG_EEPROM_AT24
-#if DT_INST_0_ATMEL_AT24
-#define DT_INST_0_ATMEL_AT24_SPI_MAX_FREQUENCY 0
-#define DT_INST_0_ATMEL_AT24_CS_GPIOS_CONTROLLER NULL
-#define DT_INST_0_ATMEL_AT24_CS_GPIOS_PIN 0
-#ifndef DT_INST_0_ATMEL_AT24_WP_GPIOS_CONTROLLER
-#define DT_INST_0_ATMEL_AT24_WP_GPIOS_CONTROLLER NULL
-#define DT_INST_0_ATMEL_AT24_WP_GPIOS_PIN 0
-#define DT_INST_0_ATMEL_AT24_WP_GPIOS_FLAGS GPIO_ACTIVE_LOW
-#endif
-EEPROM_AT2X_DEVICE(24, 0);
-#endif
+#define EEPROM_AT24_DEVICE(n) EEPROM_AT2X_DEVICE(n, 24)
+#define EEPROM_AT25_DEVICE(n) EEPROM_AT2X_DEVICE(n, 25)
 
-#if DT_INST_1_ATMEL_AT24
-#define DT_INST_1_ATMEL_AT24_SPI_MAX_FREQUENCY 0
-#define DT_INST_1_ATMEL_AT24_CS_GPIOS_CONTROLLER NULL
-#define DT_INST_1_ATMEL_AT24_CS_GPIOS_PIN 0
-#ifndef DT_INST_1_ATMEL_AT24_WP_GPIOS_CONTROLLER
-#define DT_INST_1_ATMEL_AT24_WP_GPIOS_CONTROLLER NULL
-#define DT_INST_1_ATMEL_AT24_WP_GPIOS_PIN 0
-#define DT_INST_1_ATMEL_AT24_WP_GPIOS_FLAGS GPIO_ACTIVE_LOW
-#endif
-EEPROM_AT2X_DEVICE(24, 1);
-#endif
+#define DT_INST_AT2X_FOREACH(t, inst_expr) \
+	UTIL_LISTIFY(DT_NUM_INST(atmel_at##t), DT_CALL_WITH_ARG, inst_expr)
```

Comment:
Should likely wait for https://github.com/zephyrproject-rtos/zephyr/pull/23416

Reply:
No, it can’t. This driver supports multiple, different compatibles.